test/cases/enum.ts

```diff
@@ -6,7 +6,8 @@ export var schema = {
       "enum": ["a", "b", "c"]
     },
     "bar": {
-      "enum": [1, 2, 3]
+      "enum": [1, 2, 3],
+      "tsEnumNames": ["One","Two","Three"]
```

Comment:
I think the expected behavior is not this. If useTypescriptEnums=true:

1. If `enum` is a list of strings, then compile to a TS enum
2. If `enum` is a list of strings and the key `enumValues` exists and is a list of numbers, use `enumValues` as the values for the keys defined in `enum`
3. If `enum` is not a list of strings, throw a TypeError
4. If `enumValues` is defined and is not a list of numbers, throw a TypeError